### ISSUE TITLE
Can BLIP2 work with small batch size?

### ISSUE THREAD

**Issue by yxchng** [salesforce/LAVIS#273]
Can BLIP2 work with small batch size like 64,128,256?

**Comment by dxli94** [comment]
Pre-training with small batch sizes will likely to lead diminished effect of ITC loss.

Finetuning with smaller batch sizes is possible, depending on your task and dataset.

**Comment by yxchng** [comment]
1. Is there any good ways to reduce memory usage? I am short on memory and can only train with small batch size. 
2. How important is ITC loss? Will I still get decent result with diminished effect of ITC loss?

**Comment by dxli94** [comment]
Depending on what do you mean by "decent",  ITC with small batch sizes hurts image-text alignment thus performance.

**Comment by ldfandian** [comment]
I tried to pretrain (stage1) blip-2 network on my single A100 machine and found any batch_size larger than 400 leading to GPU OOM error.

Anything wrong with my config?
